invenio_app_ils/document_requests/loaders/jsonschemas/document_request_document.py

```diff
@@ -5,18 +5,18 @@
 # invenio-app-ils is free software; you can redistribute it and/or modify
 # it under the terms of the MIT License; see LICENSE file for more details.
 
-"""Invenio App ILS Document Request Pending loader JSON schema."""
+"""Invenio App ILS Document Request Document loader JSON schema."""
 
 from invenio_records_rest.schemas import RecordMetadataSchemaJSONV1
 from marshmallow import EXCLUDE, fields
 
 
-class DocumentRequestPendingSchemaV1(RecordMetadataSchemaJSONV1):
-    """Document Request Pending schema."""
+class DocumentRequestDocumentSchemaV1(RecordMetadataSchemaJSONV1):
+    """Document Request Document schema."""
 
     class Meta:
         """Meta attributes for the schema."""
 
         unknown = EXCLUDE
 
-    remove_fields = fields.List(fields.Str())
+    document_pid = fields.Str()
```

Comment:
This is not mandatory because on delete there is no `document_pid` payload, right?

Reply:
Exactly, if we want to be super conservative, in a previous commit, I was actually submitting the document_pid and I was checking if it matches the one we are trying to remove. We can go down either path wdyt?

Reply:
I think it is good enough! :)
I would maybe add a comment that on POST you have it, and on DELETE no.

Reply:
@topless what about if you use [partial](https://marshmallow.readthedocs.io/en/stable/quickstart.html#partial-loading)? You can set it to required here, so that in the `POST` method it is automatically checked and you can get rid of your own validation, and in the `DELETE`, use `partial=True`.

Reply:
@ntarocco I have made the `document_pid` required, and now I am submitting it also in the `DELETE` request. The main reason for that is that I didn't find a simple way to pass `partial=True` to the load method. As far I understood this is the place where the `load()` is called: https://github.com/inveniosoftware/invenio-records-rest/blob/526dcaef46dfe79acf737feee4570fd6e0b25679/invenio_records_rest/loaders/marshmallow.py#L91. According to the documentation for partial it should be something like `.load(request_json, partial=True). In invenio-records-rest we don't allow extra params to be passed down to load function. There are ways to fix/cheat this around, I am looking forward for you input cause I might have missed something in the process.

Reply:
As it concerns, submitting data with `DELETE` there is a worth reading SO question https://stackoverflow.com/questions/299628/is-an-entity-body-allowed-for-an-http-delete-request